mobile/src/components/my-courses/components/courses-as-mentor/components/students-count-cell/students-count-cell.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FC } from 'react';
+
+import { AppColor } from '~/common/enums/enums';
+import { CourseUpdateMentoringDto } from '~/common/types/types';
+import { Icon, Input, Pressable, View } from '~/components/common/common';
+import { getNameOf } from '~/helpers/helpers';
+import { useAppDispatch, useAppForm, useEffect } from '~/hooks/hooks';
+import { myCoursesActions } from '~/store/actions';
+import { courseMentoringUpdateCount } from '~/validation-schemas/validation-schemas';
+
+import { styles } from './styles';
+
+type Props = {
+  studentsCount: number;
+  courseId: number;
+};
+
+const StudentsCountCell: FC<Props> = ({ courseId, studentsCount }) => {
+  const dispatch = useAppDispatch();
+
+  const { control, errors, handleSubmit, setValue } =
+    useAppForm<CourseUpdateMentoringDto>({
+      defaultValues: { courseId: courseId, studentsCount: undefined },
+      validationSchema: courseMentoringUpdateCount,
+    });
+
+  const hitSlop = { top: 5, bottom: 5, left: 5, right: 5 };
+
+  const handleEditCourse = (course: CourseUpdateMentoringDto): void => {
+    dispatch(myCoursesActions.updateCoursesMentoring(course));
+  };
+
+  useEffect(() => {
+    setValue(
+      getNameOf<CourseUpdateMentoringDto>('studentsCount'),
+      studentsCount.toString(),
+    );
```

Comment:
use reset instead of setValue

Reply:
In that way  i have a types dismatch, because to submit form I need pass studentsCount as number, and to set its default value it must be string

Reply:
you can convert types where it's needed

Reply:
Please give me a hit, I can't find the way to avoid warnings and make request sent =))

Reply:
give an example of your code and warning

Reply:
And also, could you please explain why we shouldn't use setValue? I want to know, to avoid such solutions in future

Reply:
Seems. I've solved it, while writing you the code example))